### DISCUSSION TITLE
Connection String Deadline

### DISCUSSION THREAD

**Discussion by JamiuKolawole04** [supabase/supabase#20388] [Questions] [answered]
Hey guys, I received a mail from supabase telling me connection string update deadline.The issue here is that I am accessing the db from backend direcltly with knexjs.Knexjs doesn't request fro connection string.It was using password, host, client, and databse name for connection.Does this affect me because I have contacted supase support team but I was told support is not available..  Should I also change my connections please? anyone with the same issue??

**Comment by filipecabaco** [comment]
Hi

From what I can gather from the docs you should be able to give it the connection string for your database:
https://knexjs.org/guide/#node-js

<img width="768" alt="Screenshot 2024-01-12 at 16 40 44" src="https://github.com/supabase/supabase/assets/1697301/3960760b-dd73-4938-afda-ba433c044fd0">

Did you try this method?

**Comment by JamiuKolawole04** [comment]
Yeah.I was using that method but not that approach. I was not connecting directly with the connection string.Instead I am connecting with  the port, host and password. like that pic below
![Screenshot (228)](https://github.com/supabase/supabase/assets/75920331/c85083ec-2e97-4d98-8cf9-ca2d3facfbb7)

**Reply by filipecabaco** [reply]
if you use the connection string only with the Supavisor url are you able to connect?

**Comment by JamiuKolawole04** [comment]
I haven't used the connection string before.I use only the configurations

**Reply by filipecabaco** [reply]
Ok I will try the lib with a project against supavisor and report back to see if this will be an issue 👀

**Reply by filipecabaco** [reply, marked as answer]
Tested against a project using supavisor and worked as expected when queried `SELECT 1;`
<img width="299" alt="Screenshot 2024-01-15 at 17 33 28" src="https://github.com/supabase/supabase/assets/1697301/9c7c665b-828a-486b-b9cd-861e5822d442">
<img width="391" alt="Screenshot 2024-01-15 at 17 33 55" src="https://github.com/supabase/supabase/assets/1697301/cc99b23a-f248-4c54-bf28-d218d43a7a0b">
<img width="413" alt="Screenshot 2024-01-15 at 17 34 10" src="https://github.com/supabase/supabase/assets/1697301/86d80361-7c3a-4a36-9084-dea6b93b33bd">


This means you should expect everything to work as expected 👍

**Comment by JamiuKolawole04** [comment]
ok.Thanks
